src/test-run/browser-manipulation-queue.js

```diff
@@ -1,75 +1,87 @@
-import { EventEmitter } from 'events';
 import shortId from 'shortid';
-import OS from 'os-family';
-import { resize as resizeWindow, getViewportSize } from 'testcafe-browser-natives';
+import { getViewportSize, generateThumbnail } from 'testcafe-browser-natives';
 import { isServiceCommand } from './commands/utils';
 import COMMAND_TYPE from './commands/type';
-import Warning from '../warnings';
+import WarningsLog from '../warnings/log';
 import WARNING_MESSAGE from '../warnings/message';
 
-export default class BrowserManipulationQueue extends EventEmitter {
-    constructor (windowId, screenshotCapturer) {
-        super();
 
+const SCREENSHOT_MANIPULATION_DESCRIPTION = 'take a screenshot';
+const RESIZE_MANIPULATION_DESCRIPTION     = 'do a window resize';
+
+export default class BrowserManipulationQueue {
+    constructor (browserConnection, screenshotCapturer) {
         this.commands           = [];
-        this.windowId           = windowId;
+        this.windowId           = browserConnection.id;
+        this.browserProvider    = browserConnection.provider;
         this.screenshotCapturer = screenshotCapturer;
+
+        this.warningLog = WarningsLog.get('manipulation:' + this.windowId);
```

Comment:
I would like to keep warnings system as simple as possible. So far all warnings are isolated in BrowserManipulationQueue it would be nice to keep them there as just events. Regarding debugging warnings - they should be implemented separately and shouldn't be leaked to providers (implementers should choose their own logging implementation)